src/buildercore/trop.py

```diff
@@ -94,44 +97,58 @@ def rds_security(context):
 #
 
 
-def instance_tags(context):
-    return [
-        ec2.Tag('Name', context['stackname']),
+def instance_tags(context, node=None):
+    tags = [
         ec2.Tag('Owner', context['author']),
+        # hierarchically ordered
         ec2.Tag('Project', context['project_name']),
+        ec2.Tag('Cluster', context['stackname']),
     ]
-
-def ec2instance(context):
+    if node:
+        tags.append(ec2.Tag('Name', '%s--%d' % (context['stackname'], node)))
+        tags.append(ec2.Tag('Node', '%s' % node))
+    else:
+        tags.append(ec2.Tag('Name', context['stackname']))
+    return tags
+
+def ec2instance(context, node):
     lu = partial(utils.lu, context)
+    build_vars = dict(context)
```

Comment:
we create `context in `cfngen`, so it should always be a dict (this does a copy so that we can add stuff to the single node's context.) Time for an assertion?